\documentclass[12pt]{article}
\usepackage{geometry}                % See geometry.pdf to learn the layout options. There are lots.
\geometry{a4paper,
 total={170mm,257mm},
 left=20mm,
 top=20mm,
 bottom=40mm}                   % ... or a4paper or a5paper or ... 
%\geometry{landscape}                % Activate for for rotated page geometry
\usepackage[parfill]{parskip}    % Activate to begin paragraphs with an empty line rather than an indent
\usepackage{graphicx}
\usepackage{amssymb}
\usepackage{epstopdf}
\usepackage{float}
\usepackage{hyperref}
\hypersetup{%
  colorlinks=true,% hyperlinks will be coloured
  linkcolor=blue,% hyperlink text will be green
}
\DeclareGraphicsRule{.tif}{png}{.png}{`convert #1 `dirname #1`/`basename #1 .tif`.png}

\usepackage[backend=bibtex, sorting=none, style=chicago-authordate]{biblatex}
\setlength\bibitemsep{\baselineskip}
\addbibresource{exampleRef.bib}

%% LOGOS
\usepackage{fancyhdr}
\pagestyle{fancyplain} % use fancy for all pages except chapter start
\lhead{\includegraphics[height=1.3cm, width=2.2cm]{Logos/NBIS-logo-orange-text.png}} % left logo
\rhead{\includegraphics[height=1.3cm, width=4cm]{Logos/SciLifeLab-logo.jpg}} % right logo
\renewcommand{\headrulewidth}{0pt} % remove rule below header

%% DEFINE VARIABLES
\newcommand{\staff}{Olga Dethlefsen}
\newcommand{\staffEmail}{olga.dethlefsen@nbis.se}
\newcommand{\staffWeb}{http://nbis.se/about/staff/olga-dethlefsen/}

\newcommand{\coordinator}{}

\newcommand{\affilations}{National Bioinformatics Infrastructure Sweden, Science for Life Laboratory, Stockholm University}
\newcommand{\supportWeb}{http://nbis.se/support/support.html}
\newcommand{\uppmaxWeb}{http://www.uppmax.uu.se/support/faq/general-miscellaneous-faq/acknowledging-uppmax--snic--and-uppnex/}

\newcommand{\noIssue}{\#9999}
\newcommand{\noUppmax}{b29999}

\newcommand{\fastqc}{\texttt{FastQC/0.11.2}}
\newcommand{\trimmomatic}{\texttt{trimmomatic/0.32}}
\renewcommand{\star}{\texttt{star/2.4.1c}}
\newcommand{\featureCounts}{\texttt{featureCounts/1.5.0}}
\newcommand{\subread}{\texttt{subread/1.4.5}}
\newcommand{\samtools}{\texttt{samtools/0.1.19}}
\newcommand{\multiqc}{\texttt{MultiQC/0.3.1}}
\newcommand{\homer}{\texttt{homer/4.7.2}}

\newcommand{\refGenome}{\texttt{Mus\_musculus.GRCm38.\-dna.primary\_assembly.fa}}
\newcommand{\refAnnotation}{\texttt{Mus\_musculus.GRCm38.81.gtf}}
\newcommand{\refSource}{\texttt{http://www.ensembl.org/index.html}}


%% BEGIN DOCUMENT
%\date{}
\begin{document}
\pagestyle{fancy}
\SweaveOpts{concordance=TRUE}

%% TITLE PAGE
\title{Example RNA-seq report}
\author{}
\maketitle
\thispagestyle{fancy}

\vspace{2cm}
\begin{center}
\begin{tabular}{l r}
Issue number: & {\noIssue} \\
Request by: & Jan User <jan.user@ki.se> \\ 
Principal Investigator: & Maria Investigator <maria.investigator@ki.se> \\
Organisation: & Karolinska Insitutet \\
NBIS staff: & {\staff}<{\staffEmail}>
\end{tabular}
\end{center}

%% TABLE OF CONTENTS
\newpage
\tableofcontents

%% SUPPORT REQUEST
\newpage
\section{Support request}

To answer our question we have performed RNA-seq, total RNA ribosomal RNA depleted. We had 4 groups and would like help to finding differentially expressed genes

\section{Important practical information}
\subsection{Data responsibility}
\label{DataRes}
\footnotesize
\begin{description}
\item [NBIS \& Uppnex] Unfortunately, we do not have resources to keep any files associated with the support request. We kindly suggest that you store safely the results delivered by us. In addition, we kindly ask that you remove the files from UPPMAX/UPPNEX. The main storage at UPPNEX is optimized for high-speed and parallel access, which makes it expensive and not the right place for longer time archiving. Please consider others by not taking up the expensive space.
  \item[Long-term backup] The responsibility for data archiving lies with universities and we recommend asking your local IT for support with long-term data archiving. Also a newly established \href{https://www.scilifelab.se/data/}{Data Office} at SciLifeLab may be of help to discuss other options. 
  %\item[Data submission] NBIS can help with raw sequencing data submission to ENA. Please contact our data manager, Niclas Jareborg, niclas.jareborg@nbis.se for more details. 
  \end{description}

\subsection{Acknowledgments}
\label{acknowledgments}
If you are presenting the results in a paper, at a workshop or conference, we kindly ask you to acknowledge us.
\begin{description}
\item[NBIS staff] are encouraged to be co-authors when this is merited in accordance to the ethical recommendations for authorship, e.g. {\href{http://www.icmje.org/recommendations/browse/roles-and-responsibilities/defining-the-role-of-authors-and-contributors.html}{ICMJE recommendations}}. If applicable, please include {\href{\staffWeb}{{\staff}, \affilations}} as co-author. In other cases, NBIS would be grateful if support by us is acknowledged in publications according to this example: \href{https://bils.se/resources/support.html}{"Support by NBIS (National Bioinformatics Infrastructure Sweden) is gratefully acknowledged."}
\item[Uppmax] kindly asks you to acknowledge UPPMAX and SNIC. If applicable, please add: {\href{\uppmaxWeb}{The computations were performed on resources provided by SNIC through Uppsala Multidisciplinary Center for Advanced Computational Science (UPPMAX) under Project SNIC {\noUppmax}.}}
% \item[Mosler] if applicable, according to the resource user agreement for the Mosler system for sensitive data, please acknowledge Mosler by adding \href{https://nbis.se/infrastructure/mosler.html}{Support by NBIS (National Bioinformatics Infrastructure Sweden) for the use of the Mosler system for sensitive data is gratefully acknowledged}


\item[NGI Stockholm] In any and all publications based on data from NGI Sweden, the authors must acknowledge SciLifeLab, NGI and Uppmax, like so: \href{https://ngisweden.scilifelab.se/info/faq#how-do-i-acknowledge-ngi-in-my-publication}{The authors would like to acknowledge support from Science for Life Laboratory, the National Genomics Infrastructure, NGI, and Uppmax for providing assistance in massive parallel sequencing and computational infrastructure.}
\end{description}
%% WORK LOG SECTION
\section{Work log}
A brief project history containing key points
\begin{description}
\item[2015-09-15] first meeting with Jan to discuss experimental design, available data and desired results.	As first results, Jan would like to receive lists of differentially expressed (DE) genes between the two time points for the two groups
\item[2015-10-09] meeting with Jan to go over the DE results. Agreed that Jan will go over the DE results and try running gene set enrichment analyses using DAVID website.
\item[2015-11-06] run and emailed Jan the exon usage results for the 4 comparisons
\item[2015-12-01] run and emailed Jan the motif discovery Homer results
\item[2015-12-10] meeting with Jan to discuss the additional 
\item[2015-12-20] final results and report delivered
\end{description}


%% MATERIALS AND METHODS SECTION
\section{Materials and Methods}
\subsection{Available data}
Data were delivered to Inbox on Uppnex {\noUppmax} in fastq format using Illumina 1.8 quality scores. Data were from a paired-end run, with one file for the forward reads and one file for the reverse reads following a naming convention: [LANE\_[DATE]\_[FLOWCELL]\_[SCILIFE NAME]\_[READ].fastq.gz. 

\subsection{Data processing}
Raw sequencing reads were processed to obtain counts per genes for each samples. This included: 
\begin{enumerate}
  \item {\fastqc} [\cite{Andrews2010}] quality check on raw sequencing reads
  \item {\trimmomatic} [\cite{Bolger2014}] reads filtering for quality score and read length. Reads with average quality below 20 (within 4-base wide sliding window) and/or shorter than 36 bases were removed
  \item {\star} [\cite{Dobin2013}] was used to align the reads to the reference genome {\refGenome} using the annotation {\refAnnotation}, with reference genome and annotation downloaded from {\refSource}
  \item {\featureCounts} from {\subread} \cite{Liao2013}] was used to count the fragments in the exon regions as defined in the {\refAnnotation} file, using default parameters. Specifically, for paired-end reads, a fragment is said to overlap a feature if at least one read base is found to overlap the feature. Fragments overlapping with more than one feature and multi-mapping reads are not counted
  \item Counts from multiple lanes were added, if applicable
  \item {\samtools}[\cite{Li2009}] were used to sort and index the BAM files containing the aligned reads, e.g. for visualization in \href{https://www.broadinstitute.org/igv/}{IGV} genome browser 
  \item {\multiqc}  [\cite{Ewels2016}] was used to aggregate results from {\fastqc}, {\star} and {\featureCounts} across many samples into a single report
\end{enumerate}

\subsection{Differential expression}
All analyses were performed under \texttt{R}, a programming language and software environment for statistical computing and graphics. Details on the \texttt{R} version and packages used can found at the end of this document in \nameref{sessionInfo}
\begin{enumerate}
  \item \texttt{biomaRt} package was used to annotate Ensembl gene identifiers with chromosome name, official gene symbol and description. 
  \item low count reads were filtered by keeping reads with at least 1 read per million in at least 2 samples
  \item \texttt{edgeR} package was used to normalize for the RNA composition by finding a set of scaling factors for the library sizes that minimize the log-fold changes between the samples for most genes, using a trimmed mean of M values (TMM) between each pair of samples.
  \item the normalized counts were used to examine the samples for outliers and relationships, using Multidimensional Scaling and heatmap based on the Pearson correlation coefficient between every sample pair
    \item the normalized counts were used to examine the samples for outliers and relationships, using Multidimensional Scaling and heatmap based on the Pearson correlation coefficient between every sample pair
    \item \texttt{edgeR} package was to define design matrix based on the experimental design, fitting gene-wise glms model and conducting likelihood ratio tests for the selected group comparisons
\end{enumerate}

\subsection{Exon usage}
\begin{enumerate}
  \item \texttt{DEXseq} package in \texttt{R} was used to infer differential exon usage
  \item the provided with the \texttt{DEXseq} package \texttt{Python} scripts were used to prepare a flattened GTF file based on the {\refAnnotation} and to obtain counts per each exon given the aligned BAM files
  \item size factors measuring the relative sequencing depth were estimated to adjust for coverage biases
  \item variability of the data was then estimated to be able to distinguish technical and biological variation from real effects on exon usage due to the different conditions. Briefly, per-exon dispersions are calculated using a Cox-Reid adjusted profile likelihood estimation, then a dispersion-mean relation is fitted to this individual dispersion values and finally, the fitted values are taken as a prior in order to shrink the per-exon estimates towards the fitted values
  \item having the dispersion estimates and the size factors, differential exon usage was tested. For each gene, \texttt{DEXSeq} fits a generalized linear model with the formula \( \sim samples + exon + condition:exon\) and compares it to the null model  \( \sim samples + exon\). The deviances of both fits are compared using a $\chi^2$-distribution, giving rise to a p value, indicative whether the null model is sufficient to explain the data or whether it may be rejected in favor of the alternative containing an interaction coefficient for condition$\colon$exon. The latter means that the fraction of the gene's reads that fall onto the exon under the test differs significantly between the experimental conditions.
  \item the obtained p-values were BH adjusted for multiple comparison
\end{enumerate}

\subsection{HOMER Motif Analysis}
\begin{enumerate}
  \item {\homer} was used to analyze the promoters of genes and look for motifs that are enriched in the target gene promoters relative to other promoters. The analyses followed the \href{http://homer.salk.edu/homer/motif/index.html}{'Analyzing lists of genes with promoter motif analysis' tutorial}
  \item briefly, for each comparison, the analyses were run for the differentially expressed genes, separately for down- and up-regulated genes, were differentially expressed genes were defined at 5\% FDR and absolute minimum log 2 fold change of 1. 
  \item the analyses included Gene Ontology enrichment calculations, \textit{de novo} motif analysis and known motif enrichment analysis
\end{enumerate}



\section{Results}
\subsection{Data exploration}
Table count, containing counts measured across many genes and samples, is a typical example of a multidimensional dataset, where \textit{N} objects (samples) were measured on \textit{p} numeric variables. Hence, to examine the samples for outliers and other relationship one can look at the several multivariate techniques that aim to reduce the dataset dimensions and to reveal the data structure by plotting samples in one or two dimensions. Multidimensional Scaling provides a visual representation of the pattern of proximities among a set of objects, here samples. Another method of visualization is a heatmap based on the Pearson correlation coefficient, calculated between every sample pair. 

% INPUT
<<input, echo=F>>=
# Input
  library(edgeR)
  library(ggplot2)
  library(gplots)
  library(biomaRt)
  library(xtable)
  #library(DEXSeq)

  dir.data <- "DATA"
  file.counts <- "DATA/count_table.txt"
  file.annotations <- "DATA/count_table_annotations.txt"

  # Preparing samples group labels
    sample.time <- rep(c('time1','time2'), each=8)
    sample.treatment <- rep(rep(c('GR1','GR2'), each=4), 2)
    sample.groups <- factor(paste(sample.treatment, sample.time, sep='.'))
    
  # Selecting colour for each group 
    colors.groups <- as.character(sample.groups)
    colors.groups[colors.groups=="GR1.time1"]="darkorange1"
    colors.groups[colors.groups=="GR1.time2"]="firebrick3"
    colors.groups[colors.groups=="GR2.time1"]="cyan2"
    colors.groups[colors.groups=="GR2.time2"]="blue2"
    
# Reading in data
  table.counts <- as.matrix(read.delim(file.counts))    
  table.annotations <- as.matrix(read.delim(file.annotations))
@ 

% BUILDING edgeR object, filtering and normalizing, preparing MDS frame
<<QC, echo=F, include=FALSE>>=
  #building edgeR object
  data.cds <- DGEList(table.counts, group = sample.groups)
  
  # print effective library sizes
  #print(data.cds$samples$lib.size * data.cds$samples$norm.factors)
  
  # filtering low count reads (keeping rads with at least 1 read per million in at least 2 samples)
  data.cds <- data.cds[rowSums(1e+06 * data.cds$counts/expandAsMatrix(data.cds$samples$lib.size, dim(data.cds)) > 1) >= 2, ]
  
  # normalizing data
  data.cds <- calcNormFactors(data.cds, method='TMM')
  data.norm <- cpm(data.cds, prior.count=2, log=TRUE)

  # 4 MDS plot
  mds <- plotMDS(data.norm, top=dim(data.norm)[1], main = "MDS Plot", labels = colnames(data.cds$counts))
  mds.frame <- data.frame(x=mds$x, y=mds$y, group=sample.groups, treatment=sample.treatment, time=sample.time)

  mds1 <- ggplot(mds.frame, aes(x,y, color=sample.groups))+geom_point(size=4)+xlab("Leading logFC dim 1")+ylab("Leading logFC dim2")+theme(legend.title = element_blank(), text = element_text(size=12))
  mds2 <- ggplot(mds.frame, aes(x,y, color=row.names(mds.frame)))+geom_point(size=4)+xlab("Leading logFC dim 1")+ylab("Leading logFC dim2")+theme(legend.title = element_blank(), text = element_text(size=12))
  
  # 4 heatmap
  # Pairwise Pearson correlation between samples    
  cor.matrix <- matrix(data=NA, nrow=ncol(data.norm), ncol=ncol(data.norm))
    for (i in 1:ncol(data.norm)){
      for (j in 1:ncol(data.norm))
      { cor.matrix[i,j] <- cor(data.norm[,i], data.norm[,j])}}
    rownames(cor.matrix) <- colnames(data.norm)
    colnames(cor.matrix) <- colnames(data.norm)
   
  #print(head(mds.frame))
@

\begin{figure}[H]
\begin{center}
<<label=fig1,fig=TRUE, echo=FALSE, width=7, height=5>>=
print(mds1)
@
\end{center}
\caption{MDS plot on the normalized and filtered counts colour-coded by samples groups}
\end{figure}  

\begin{figure}[H]
\begin{center}
<<label=fig2,fig=TRUE, echo=FALSE, width=7, height=5>>=
print(mds2)
@
\end{center}
\caption{MDS plot on the normalized and filtered counts colour-coded by individual samples}
\end{figure} 

\begin{figure}[H]
\begin{center}
<<label=fig3,fig=TRUE, echo=FALSE, width=7, height=7>>=
  heatmap.2(cor.matrix, ColSideColors = colors.groups, dendrogram = "column", trace='none', scale='none', margins = c(6,6))
@
\end{center}
\caption{Heatmap based on the pair-wise Pearson correlation coefficient between samples}
\end{figure} 

\subsection{Differential expression} 
<<DEdesign, echo=F>>=
  # design matrix for GLM model  
  my.design <- model.matrix(~0+sample.groups)
  colnames(my.design) <- levels(data.cds$samples$group)
  
  # estimating dispersion
  data.cds <- estimateDisp(data.cds, my.design, trend.method='movingave')
  
  # defining contrasts
  my.contrasts <- makeContrasts(
      GR1=GR1.time2-GR1.time1,
      GR2=GR2.time2-GR2.time1,
      levels=my.design
    )
  
  # fitting genewise glm model
    fit <- glmFit(data.cds, my.design)
@

\subsubsection{Estimating BCVs}
Two levels of variation, technical and biological, can be distinguished in any RNA-Seq experiment. Biological coefficient of variation (BCV) is the coefficient of variation with which the (unknown) true abundance of the gene varies between replicate RNA samples. It represents the CV that would remain between biological replicates if sequencing depth could be increased indefinitely. The technical CV decreases as the size of the counts increases. BCV on the other hand does not. BCV is therefore likely to be the dominant source of uncertainty for high-count genes, so reliable estimation of BCV is crucial for realistic assessment of differential expression in RNA-Seq experiments. \texttt{edgeR} uses empirical Bayes methods that permit the estimation of gene-specific biological variation, even for experiments with minimal levels of biological replication.

\begin{figure}[H]
\begin{center}
<<label=fig4, fig=TRUE, echo=FALSE, width=7, height=7>>=
   plotBCV(data.cds)
@
\end{center}
\caption{Biological coefficient of variation plot showing the dispersion estimates}
\end{figure} 

<<DEfunction, echo=F>>=
function.de <- function(inp.data, inp.annotations, inp.fit, inp.contrast)
    {
      lrt <- glmLRT(fit, contrast=my.contrasts[,inp.contrast])
      lrt.top <- topTags(lrt, n=Inf, adjust.method='BH')
      lrt.table <- lrt.top$table  
    
      # Reordering the table to match the original row order
      idx <- match(rownames(inp.data), rownames(lrt.table))
      out.table <- lrt.table[idx,]
      colnames(out.table) <- paste(inp.contrast, colnames(out.table), sep='.')
      
      # Adding gene annotations
      idx <- match(rownames(out.table), inp.annotations[,1])
      out.table <- cbind(inp.annotations[idx,], out.table)
      
      # Summary of de
      de.summary <- t(summary(de <- decideTestsDGE(lrt, lfc=1)))
      colnames(de.summary) <- c("Down-regulated","Non-significant","Up-regulated")
      rownames(de.summary) <- c("Genes")
      
      # Smear plot
      plotSmear(lrt)
      
      # Out
      de.list <- list()
      de.list[[1]] <- out.table
      de.list[[2]] <- de.summary
      de.list[[3]] <- lrt
      de.list
    }  
@      

\subsubsection{GR1: time2 vs. time1}

<<de1, echo=F, results=tex, include=F>>=
de1 <- function.de(data.cds$counts, table.annotations, fit, "GR1")
tmp.de <- de1[[1]]
tmp.summary <- de1[[2]]
tmp.lrt <- de1[[3]]

# Number of down- and up-regulated
print(xtable(tmp.summary, digits=0,
      caption = "Number of down- and up-regulated differentially expressed genes given 5\\% FDR and absolute minimum log-fold-change of 1 "),
      caption.placement = "bottom")

# Top 10 changes
o <- order(tmp.de[,10])
table.out <- tmp.de[o[1:10],c(-2,-3,-5,-7,-8,-9)]
print(xtable(table.out, align=c("c","c","c","c","c"), display=c("s","s","s","f","E"),
      caption = "10 genes with the smallest FDR values"),
      caption.placement = "bottom",
      size="footnotesize",
      include.rownames=FALSE,
      floating=TRUE,
      table.placement="H")

@

\begin{figure}[H]
\begin{center}
<<label=fig5, fig=TRUE, echo=FALSE, width=7, height=7>>=
  de <- decideTestsDGE(tmp.lrt, lfc=1)
  detags <- rownames(data.cds)[as.logical(de)]
  plotSmear(tmp.lrt, de.tags=detags)
  abline(h=c(-1, 1), col="blue")
@
\end{center}
\caption{Plot log-fold change against log-counts per million, with DE genes highlighted}
\end{figure} 

\begin{figure}[H]
\begin{center}
<<label=fig6, fig=TRUE, echo=FALSE, width=7, height=7>>=
  lrt.top <- topTags(tmp.lrt, n=Inf, adjust.method='BH')
  lrt.table <- lrt.top$table  
  idx.sign <- match(detags, rownames(lrt.table))
  plot(lrt.table$logFC, -log10(lrt.table$FDR), pch=19, xlab='log2 fold change', ylab="-log10 FDR")
  points(lrt.table$logFC[idx.sign], -log10(lrt.table$FDR)[idx.sign], pch=19, col="red")
  abline(v=c(-1, 1), col="blue")
  abline(h=-log10(0.05), col="blue")
@
\end{center}
\caption{Volcano plot with DE genes highlighted in red. Horizontal blue line corresponds to FDR=0.05 and vertical blue lines correspond to absolute log2 fold change of 1}
\end{figure} 

\subsubsection{GR2: time2 vs. time1}

<<de2, echo=F, results=tex, include=F>>=
de2 <- function.de(data.cds$counts, table.annotations, fit, "GR2")
tmp.de <- de2[[1]]
tmp.summary <- de2[[2]]
tmp.lrt <- de2[[3]]

# Number of down- and up-regulated
print(xtable(tmp.summary, digits=0,
      caption = "Number of down- and up-regulated differentially expressed genes given 5\\% FDR and absolute minimum log-fold-change of 1 "),
      caption.placement = "bottom")

# Top 10 changes
o <- order(tmp.de[,10])
table.out <- tmp.de[o[1:10],c(-2,-3,-5,-7,-8,-9)]
print(xtable(table.out, align=c("c","c","c","c","c"), display=c("s","s","s","f","E"),
      caption = "10 genes with the smallest FDR values"),
      caption.placement = "bottom",
      size="footnotesize",
      include.rownames=FALSE,
      floating=TRUE,
      table.placement="H")

@

\begin{figure}[H]
\begin{center}
<<label=fig7, fig=TRUE, echo=FALSE, width=7, height=7>>=
  de <- decideTestsDGE(tmp.lrt, lfc=1)
  detags <- rownames(data.cds)[as.logical(de)]
  plotSmear(tmp.lrt, de.tags=detags)
  abline(h=c(-1, 1), col="blue")
@
\end{center}
\caption{Plot log-fold change against log-counts per million, with DE genes highlighted}
\end{figure} 

\begin{figure}[H]
\begin{center}
<<label=fig8, fig=TRUE, echo=FALSE, width=7, height=7>>=
  lrt.top <- topTags(tmp.lrt, n=Inf, adjust.method='BH')
  lrt.table <- lrt.top$table  
  idx.sign <- match(detags, rownames(lrt.table))
  plot(lrt.table$logFC, -log10(lrt.table$FDR), pch=19, xlab='log2 fold change', ylab="-log10 FDR")
  points(lrt.table$logFC[idx.sign], -log10(lrt.table$FDR)[idx.sign], pch=19, col="red")
  abline(v=c(-1, 1), col="blue")
  abline(h=-log10(0.05), col="blue")
@
\end{center}
\caption{Volcano plot with DE genes highlighted in red. Horizontal blue line corresponds to FDR=0.05 and vertical blue lines correspond to absolute log2 fold change of 1}
\end{figure} 


% Putting DE results together and saving files
<<final, echo=F>>=
  # Normalized data and corresponding counts
  idx <- match(rownames(data.norm), table.annotations[,1])
  norm_table_annotations <- table.annotations[idx,]
  write.table(data.norm, paste(dir.data, 'norm_table.txt', sep='/'), sep='\t', quote=F, row.names=T)
  write.table(norm_table_annotations, paste(dir.data, 'norm_table_annotations.txt', sep='/'), sep='\t', quote=F, row.names=F)
  
  # Combined DE results
  de.all <-cbind(de1[[1]], de2[[1]][,6:ncol(de2[[1]])])
    write.table(de.all, paste(dir.data, 'DE.txt', sep='/'), sep='\t', quote=F, row.names=F)
@  

%% WHAT HAS BEEN DELIVERED
\section{Deliverable}
Below is the list of key tab-delimited text files containing the key results from the described data analyses. All results incl. computational scripts can be found on Uppnex under project \texttt{{\noUppmax}}. 
<<Output, echo=F>>=
print(dir(dir.data, pattern=".txt"))
@    
where, 

\begin{description}
  \item[DE.txt] contains the differential expression results for all the analyses
  \item[count\_table.txt] contains the raw genes counts
  \item[count\_table\_annotations.txt] contains the annotations for the genes in the count\_table.txt
  \item[norm\_table.txt] contains filtered and normalized genes expression values (TMM)
  \item[norm\_table\_annotations.txt] contains the annotations for the norm\_table.txt
\end{description}

\section{R session info}
\label{sessionInfo}
%' and the number of samples is
<<noGenes, echo=F>>=
sessionInfo()
@ 

\section{Where to go next}
There is a wide selection of online user-friendly tools available for investigating the interesting genes or list of DE genes. Few recommended below 
\begin{description}
  \item[\href{hhttp://biit.cs.ut.ee/clustvis/}{ClustVist}] for creating Principal Component Analysis plots and heatmaps
  \item[\href{http://bioinfogp.cnb.csic.es/tools/venny/}{Venny}] helps to prepare Venn diagram showing relations between a finite collection of different sets, e.g. between list of DE genes from different comparisons
  \item[\href{https://david.ncifcrf.gov}{DAVID}] for a comprehensive set of functional annotation tools for investigators to understand biological meaning behind large list of genes, including identification of enriched Gene Ontology terms, discovering enriched functional-related gene groups, visualizing genes on BioCarta \& KEGG pathway and many more
  \item[\href{http://revigo.irb.hr}{REVIGO}] for summarizing long list of Gene Ontology terms and visualization in semantic similarity-based scatter plots, interactive graphs or tag clouds
\end{description}

\section{Support project closing procedures}
You should soon be contacted by one of our managers, Jessica Lindvall <jessica.lindvall@nbis.se> or Henrik Lantz <henrik.lantz@nbis.se>, with a request to close down the project in our internal system and for invoicing matters. If we do not hear from you within \textbf{30 days} the project will be automatically closed and invoice sent. Again, we would like to remind you about data responsibility and acknowledgements, see \nameref{DataRes} and \nameref{acknowledgments}. 

You are naturally more than welcome to come back to us with further data analysis request at any time via \href{http://nbis.se/support/support.html}{http://nbis.se/support/support.html}. Thank you for using NBIS and all the best for future research. 


\newpage
\printbibliography

\end{document}  